src/main/java/qupath/ext/instanseg/core/InstanSegModel.java

```diff
@@ -251,8 +258,64 @@ public Optional<Integer> getNumChannels() {
         return getModel().flatMap(model -> Optional.of(extractChannelNum(model)));
     }
 
-    private static int extractChannelNum(BioimageIoSpec.BioimageIoModel model) {
-        int ind = model.getInputs().getFirst().getAxes().toLowerCase().indexOf("c");
+    /**
+     * Try to check the output tensors from the model spec.
+     * @return The output tensors if the model is downloaded, otherwise empty.
+     */
+    public Optional<List<OutputTensor>> getOutputs() {
+        return getModel().flatMap(model -> Optional.ofNullable(model.getOutputs()));
+    }
+
+    /**
+     * Try to check the output classes from the model spec.
+     * @return The output classes if the model is downloaded, and it's present, otherwise empty.
+     */
+    public List<String> getClasses() {
+        var config = model.getConfig().getOrDefault("qupath", null);
+        if (config instanceof Map configMap) {
+            List<String> classes = new ArrayList<>();
+            var el = configMap.get("classes");
+            if (el != null && el instanceof List elList) {
+                for (var t: elList) {
+                    classes.add(t.toString());
+                }
+            }
+            return classes;
+        }
+        return List.of();
+    }
+
+    public enum OutputType {
```

Comment:
Needs a javadoc to explain what it is, and also what each enum means.

(We also use `preferredOutputType` with a different meaning elsewhere)

Reply:
Yes, specifyign that it's a tensor type is more clear